apps/storefront/src/pages/Login/index.tsx

```diff
@@ -212,98 +215,105 @@ export default function Login(props: PageProps) {
     }
   };
 
-  const handleLoginSubmit = async (data: LoginConfig) => {
-    setLoading(true);
-    setLoginAccount(data);
-    setSearchParams((prevURLSearchParams) => {
-      prevURLSearchParams.delete('loginFlag');
-      return prevURLSearchParams;
-    });
+  const handleLoginSubmit = useB2BCallback(
+    B2BEvent.OnLogin,
+    async (dispatchOnLoginEvent, data: LoginConfig) => {
+      setLoading(true);
+      setLoginAccount(data);
+      setSearchParams((prevURLSearchParams) => {
+        prevURLSearchParams.delete('loginFlag');
+        return prevURLSearchParams;
+      });
+
+      if (isCheckout) {
+        try {
+          await loginCheckout(data);
+          window.location.href = CHECKOUT_URL;
+        } catch (error) {
+          b2bLogger.error(error);
+          getForcePasswordReset(data.emailAddress);
+        }
+      } else {
+        try {
+          const loginData = {
+            email: data.emailAddress,
+            password: data.password,
+            storeHash,
+            channelId,
+          };
+          const {
+            login: {
+              result: { token, storefrontLoginToken },
+              errors,
+            },
+          } = await b2bLogin({ loginData });
+
+          storeDispatch(setB2BToken(token));
+          customerLoginAPI(storefrontLoginToken);
+
+          dispatchOnLoginEvent(storefrontLoginToken);
```

Comment:
can we pass the data as an object with `storefrontLoginToken` as a key/value so that if we every have to extend the login event in the future we don't need to create a new event.